src/algebra/gcd_monoid/finset.lean

```diff
@@ -87,6 +87,13 @@ lcm_dvd (λ b hb, (h b hb).trans (dvd_lcm hb))
 lemma lcm_mono (h : s₁ ⊆ s₂) : s₁.lcm f ∣ s₂.lcm f :=
 lcm_dvd $ assume b hb, dvd_lcm (h hb)
 
+theorem lcm_eq_zero_iff [nontrivial α] : s.lcm f = 0 ↔ ∃ (x : β), x ∈ s ∧ f x = 0 :=
```

Comment:
I wonder if this makes sense to also state as:
```suggestion
theorem lcm_eq_zero_iff [nontrivial α] [decidable_eq α] : s.lcm f = 0 ↔ 0 ∈ finset.image f s :=
```
or
```suggestion
theorem lcm_eq_zero_iff [nontrivial α] : s.lcm f = 0 ↔ 0 ∈ f '' s :=
```

Reply:
Yes, that makes sense. I'll try it later.